src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java

```diff
@@ -76,6 +76,30 @@ default void removeEventListener(final Object... listeners)
         this.getShardCache().forEach(jda -> jda.removeEventListener(listeners));
     }
 
+    /**
+     * Adds listeners provided by the listener provider to each shard to the event-listeners that will be used to handle events.
+     * The listener provider gets a shard id applied and is expected to return a listener.
+     *
+     * <p>Note: when using the {@link net.dv8tion.jda.core.hooks.InterfacedEventManager InterfacedEventListener} (default),
+     * given listener <b>must</b> be instance of {@link net.dv8tion.jda.core.hooks.EventListener EventListener}!
+     *
+     * @param  shardedEventListenerProvider
+     *         The provider of listener(s) which will react to events.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided listener provider {@code null}.
+     */
+    default void addShardedEventListeners(final Function<Integer, Object> shardedEventListenerProvider)
```

Comment:
This is missing a `removeEventListeners(IntPredicate)`

Reply:
Could we use `IntFunction` for these?

Reply:
>This is missing a removeEventListeners(IntPredicate)

Which removes all listeners from a specific shard, or other shards? I like the idea, and I'll gladly add that if you confirm. Just making sure, because we're also missing a method that removes the listeners provided by a listener provider from all shards.